mido/midifiles/midifiles.py

```diff
@@ -373,18 +361,40 @@ def __iter__(self):
             raise TypeError("can't merge tracks in type 2 (asynchronous) file")
 
         tempo = DEFAULT_TEMPO
+        time_signature = DEFAULT_TIME_SIGNATURE
+        cum_delta = 0
         for msg in merge_tracks(self.tracks):
-            # Convert message time from absolute time
-            # in ticks to relative time in seconds.
+            # Convert relative message time to desired unit
             if msg.time > 0:
-                delta = tick2second(msg.time, self.ticks_per_beat, tempo)
+                if self.unit.lower() in ('t', 'ticks'):
+                    delta = msg.time
+                elif self.unit.lower() in ('s', 'sec', 'seconds'):
+                    delta = tick2second(msg.time, self.ticks_per_beat, tempo)
+                elif self.unit.lower() in ('b', 'beats'):
+                    delta = tick2beat(msg.time, self.ticks_per_beat,
+                                      time_signature)
+                else:
+                    raise ValueError("`unit` must be either 'ticks', 't', "
+                                     "'seconds', 's', 'beats', 'b', not %s." %
+                                     self.unit)
             else:
                 delta = 0
+            # Convert relative time to absolute values if needed
+            if self.timing.lower() in ('a', 'abs', 'absolute'):
```

Comment:
Same here. Shorthands are harder to search for help with for users etc.

Reply:
Again, an `Enum` would solve this.
No need to check for multiple strings variations.